add helper function to print top of stack after each opcode execution (#366)
scale-it/algo-builder: docs/guide/testing-teal.md

Please look at [stateful-counter.ts](https://github.com/scale-it/algo-builder/blob/master/packages/runtime/test/integration/stateful-counter.ts) to see the complete integration test suite.

## Debugging

> Based upon input from the community, we have introduced an optional parameter `debugStack` to pass with `runtime.executeTx()`. Since setting up teal debugger(start tealdbg instance, creating a dry-run dump.json) could be a lengthy process, this param is helpful to quickly  inspect the state of stack while executing TEAL code (for each opcode in smart contract in test).

Reviewer: ```suggestion
`runtime.executeTx()` function has an optional `debugStack: number` parameter. Setting up teal debugger (start tealdbg instance, creating a dry-run dump.json) could be a lengthy process, this param is helpful to quickly  inspect the state of stack while executing TEAL code (for each opcode in smart contract in test).
```

> If `debugStack` parameter is passed in `executeTx(..)`, then the TEALStack is printed to console (upto max_depth = `debugStack`) after each opcode execution in TEAL file.

Reviewer: ```suggestion
When the `debugStack: number` parameter is set, a TEAL stack is printed to a console (up to max_depth = `debugStack`) after each opcode execution in the TEAL file.
```
